proto-build/src/main.rs

```diff
@@ -256,3 +267,15 @@ fn copy_generated_files(from_dir: &Path, to_dir: &Path) {
         panic!("[error] Aborted.");
     }
 }
+
+fn copy_and_patch(src: impl AsRef<Path>, dest: impl AsRef<Path>) -> io::Result<()> {
+    let contents = fs::read_to_string(src)?;
+
+    // Patch each service definition with a feature attribute
+    let patched_contents = contents.replace(
+        TONIC_SERVICE_ATTRIBUTE,
+        &format!("{}\n{}", GRPC_FEATURE_ATTRIBUTE, TONIC_SERVICE_ATTRIBUTE),
+    );
+
+    fs::write(dest, patched_contents)
+}
```

Comment:
Hax, but unfortunately while I saw `tonic_build` lets you add attributes to generated structs/enums, it doesn't appear to provide any way to easily add an attribute to modules.

Reply:
pretty clever I like it.